### PR TITLE
Material event updates

### PR DESCRIPTION
**Pull request by Tkael**
Fix for #453.
Adds new 'MaterialTrade` and `TechnologyBroker` events

### CODE REVIEW

File: EDDI/ChangeLog.md [hunk 1]
```diff
@@ -13,9 +17,9 @@
     * First installations will now take any custom VoiceAttack installation location into account when proposing a location for EDDI.
     * Upgrade installations will continue to use whatever location was selected in the first installation.
   * Speech Responder
-    * Added 'Jet cone damage' event
+    * Added `Jet cone damage` event
```
Comment (richardbuckle) [inline EDDI/ChangeLog.md]: We haven't in the past used code quotes for event names, but I'm not averse to it.

File: Events/MaterialTradedEvent.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        public const string NAME = "Material traded";
+        public const string DESCRIPTION = "Triggered when you trade materials at a materials trader";
+        public const string SAMPLE = "{ \"timestamp\": \"2018-04-02T05:04:45Z\", \"event\": \"MaterialTrade\", \"MarketID\": 3223343616, \"TraderType\": \"encoded\", \"Paid\": { \"Material\": \"shielddensityreports\", \"Material_Localised\": \"Untypical Shield Scans \", \"Category\": \"$MICRORESOURCE_CATEGORY_Encoded;\", \"Category_Localised\": \"Encoded\", \"Quantity\": 72 }, \"Received\": { \"Material\": \"shieldfrequencydata\", \"Material_Localised\": \"Peculiar Shield Frequency Data\", \"Category\": \"$MICRORESOURCE_CATEGORY_Encoded;\", \"Category_Localised\": \"Encoded\", \"Quantity\": 2 } }";
+        public static Dictionary<string, string> VARIABLES = new Dictionary<string, string>();
+
+        static MaterialTradedEvent()
+        {
+            VARIABLES.Add("tradertype", "The type of material trader for the trade");
+            VARIABLES.Add("paid", "The name of the material lost in the trade");
+            VARIABLES.Add("paidqty", "The amount of the material lost in the trade");
```
Comment (richardbuckle) [inline Events/MaterialTradedEvent.cs]: Not keen on this abbreviation, it's too opaque for non-native speakers, would prefer "paid_quantity"

File: Events/MaterialTradedEvent.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            VARIABLES.Add("received", "The name of the material lost in the trade");
+            VARIABLES.Add("receivedqty", "The amount of the material lost in the trade");
```
Comment (richardbuckle) [inline Events/MaterialTradedEvent.cs]: Similarly would prefer "received_quantity".

File: EDDI/ChangeLog.md [hunk 4]
```diff
@@ -3,8 +3,12 @@
 ### 3.0.0-rc2
   * Core
     * The EDSM responder has been updated to send data to EDSM per their revised API. 
+    * Incorporated new material transaction events
+    * EDDI's Material Monitor will now auto-calculate maximum material limits when not otherwise defined... provided the material rarity is known.
   * Speech Responder
-    * Added 'Fighter rebuilt' event
+      * Added `Fighter rebuilt` event
+      * Added `Material trade` event
```
Comment (richardbuckle) [inline EDDI/ChangeLog.md]: Are these event names truly `code voice` for Voice Attack users? For standalone users they are just names in the Speech Responder tab.

File: Events/MaterialTradedEvent.cs (hunk 3 above)
Comment (richardbuckle) [reply Events/MaterialTradedEvent.cs]: Also shouldn't it be "gained" not "lost"?

File: Events/MaterialTradedEvent.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        [JsonProperty("tradertype")]
+        public string tradertype { get; private set; }
+
+        [JsonProperty("paid")]
+        public string paid { get; private set; }
+
+        [JsonProperty("paidqty")]
```
Comment (richardbuckle) [inline Events/MaterialTradedEvent.cs]: as above

File: Events/MaterialTradedEvent.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        public int paidqty { get; private set; }
+
+        [JsonProperty("received")]
+        public string received { get; private set; }
+
+        [JsonProperty("receivedqty")]
```
Comment (richardbuckle) [inline Events/MaterialTradedEvent.cs]: as above

File: EDDI/ChangeLog.md (hunk 4 above)
Comment (Tkael) [reply EDDI/ChangeLog.md]: Define `code voice`?

Comment (Tkael) [reply EDDI/ChangeLog.md]: The corresponding VoiceAttack commands shall be `((EDDI material traded))` and `((EDDI technology broker))` with auto-generated variables (as described at https://github.com/EDCD/EDDI/wiki/VoiceAttack-Integration#running-commands-on-eddi-events)

**Comment by Tkael** [comment]
Closing and resubmitting to allow for some code cleanup

**Pull request closed without merging**